doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+        /// Number of bytes downloaded if known
+        UInt64 BytesDownloaded;
+        /// DESIGN NOTE: BytesRequired may only be available for downloads done by Windows Package Manager itself.
+        /// Number of bytes required if known
+        UInt64 BytesRequired;
+        /// Download percentage completed
+        Single DownloadPercentage;
+        /// Install percentage if known.
+        Single InstallPercentage;
+    };
+
+    /// Result of the install
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass InstallResult
+    {
+        /// Used by a caller to correlate the install with a caller's data.
+        String CorrelationData{ get; };
+        /// Whether a restart is required to complete the install.
+        Boolean RebootRequired { get; };
+    }
+
+    /// IMPLEMENTATION NOTE: SourceOrigin from AppInstallerRepositorySource.h
+    /// Defines the origin of the app catalog details.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppCatalogOrigin
+    {
+        /// Predefined means it came as part of the Windows Package Manager package and cannot be removed.
+        Predefined,
+        /// User means it was added by the user and could be removed.
+        User,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceTrustLevel from AppInstallerRepositorySource.h
+    /// Defines the trust level of the app catalog.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppCatalogTrustLevel
+    {
+        None,
+        Trusted,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceDetails from AppInstallerRepositorySource.h
+    /// Interface for retrieving information about an app catalog without acting on it.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass AppCatalogInfo
+    {
+        /// The app catalog's unique identifier.
+        String Id { get; };
+        /// The name of the app catalog.
+        String Name { get; };
+        /// The type of the app catalog.
+        String Type { get; };
+        /// The argument used when adding the app catalog.
+        String Arg { get; };
+        /// The app catalog's extra data string.
+        String ExtraData { get; };
+        /// The last time that this app catalog was updated.
+        Windows.Foundation.DateTime LastUpdateTime { get; };
+        /// The origin of the app catalog.
+        AppCatalogOrigin Origin { get; };
+        /// The trust level of the app catalog
+        AppCatalogTrustLevel TrustLevel { get; };
+    };
+
+    /// A metadata item of a package version.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum PackageVersionMetadata
+    {
+        /// The InstallerType of an installed package
+        InstalledType,
+        /// The Scope of an installed package

```

Comment:
This is "User" and "Machine" so I could maybe re-use AppInstallScope. But the AppInstallScope could potentially eventually add a Default scope that would mean to use whatever the installer or installer yaml wants, which wouldn't make sense here so maybe it would need to be it's own enum too. If we get rid of the GetMetadata and turn them all into properties that would make more sense.

Reply:
Adding a note here as well since lower down we decided to change Machine to System in the AppInstallScope enum